utils/report.c

```diff
@@ -714,6 +716,244 @@ void report_sort_tasks(struct uftrace_data *handle, struct rb_root *name_root,
+static struct display_field *field_table[] = {
+	&field_total,
+	&field_total_avg,
+	&field_total_min,
+	&field_total_max,
+	&field_self,
+	&field_self_avg,
+	&field_self_min,
+	&field_self_max,
+	&field_call,
+};
+
+/* index of this table should be matched to display_field_id */
+static struct display_field *field_diff_table[] = {
+	&field_total_diff,
+	&field_total_avg_diff,
+	&field_total_min_diff,
+	&field_total_max_diff,
+	&field_self_diff,
+	&field_self_avg_diff,
+	&field_self_min_diff,
+	&field_self_max_diff,
+	&field_call_diff,
+};
+
+/* index of this table should be matched to display_field_id */
+static struct display_field *field_diff_full_table[] = {
+	&field_total_diff_full,
+	&field_total_avg_diff_full,
+	&field_total_min_diff_full,
+	&field_total_max_diff_full,
+	&field_self_diff_full,
+	&field_self_avg_diff_full,
+	&field_self_min_diff_full,
+	&field_self_max_diff_full,
+	&field_call_diff_full,
+};
+
+/* index of this table should be matched to display_field_id */
+static struct display_field *field_diff_full_percent_table[] = {
+	&field_total_diff_full_percent,
+	&field_total_avg_diff_full_percent,
+	&field_total_min_diff_full_percent,
+	&field_total_max_diff_full_percent,
+	&field_self_diff_full_percent,
+	&field_self_avg_diff_full_percent,
+	&field_self_min_diff_full_percent,
+	&field_self_max_diff_full_percent,
+	&field_call_diff_full_percent,
+};
+
+static void setup_default_field(struct list_head *fields, struct opts *opts,
+				struct display_field *p_field_table[])
+{
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME]);
+	add_field(fields, p_field_table[REPORT_F_CALL]);
+}
+
+static void setup_avg_total_field(struct list_head *fields, struct opts *opts,
+				  struct display_field *p_field_table[])
+{
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_AVG]);
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_MIN]);
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_MAX]);
+}
+
+static void setup_avg_self_field(struct list_head *fields, struct opts *opts,
+				 struct display_field *p_field_table[])
+{
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_AVG]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_MIN]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_MAX]);
+}
+
+void setup_field_opts(struct list_head *output_fields, struct opts *opts)
+{
+	void (*setup_default_field_func)(struct list_head*, struct opts*,
+		struct display_field *p_field_table[])
+		= &setup_default_field;
+	if (opts->avg_total) {
+		setup_default_field_func = &setup_avg_total_field;
+	} else if (opts->avg_self) {
+		setup_default_field_func = &setup_avg_self_field;
+	}
+
+	struct display_field **f_table;
+	if (opts->diff) {
+		if (opts->diff_policy && diff_policy.full) {
+			if (diff_policy.percent)
+				f_table = field_diff_full_percent_table;
+			else
+				f_table = field_diff_full_table;
+		} else
+			f_table = field_diff_table;
+	} else
```

Comment:
Please add a closing brace : https://www.kernel.org/doc/html/v5.5/process/coding-style.html#placing-braces-and-spaces